tests/unit/test_charm.py

```diff
@@ -1,18 +1,674 @@
 # Copyright 2024 Canonical Ltd.
 # See LICENSE file for licensing details.
 
+from unittest.mock import Mock, patch
+
 import ops
 import ops.testing
 import pytest
+import requests
 from charm import GocertCharm
-from scenario import Context, Event, State
+from scenario import Container, Context, Event, Network, State, Storage
 
 
 class TestCharm:
-    @pytest.fixture(scope="function", autouse=True)
+    @pytest.fixture(scope="function")
     def context(self):
         yield Context(GocertCharm)
 
-    def test_start_charm(self, context):
-        out = context.run(Event("start"), State())
+    def test_given_only_config_storage_container_cant_connect_network_not_available_gocert_not_running_when_configure_then_status_is_waiting(
+        self, context
+    ):
+        state = State(
+            storage=[Storage(name="config")],
+            containers=[Container(name="gocert", can_connect=False)],
+            networks={"juju-info": Network([], [], [])},
+            leader=True,
+        )
+
+        with patch("requests.get", return_value=Mock(side_effect=requests.ConnectionError)):
+            out = context.run(Event("config-changed"), state)
+
+        assert out.unit_status == ops.WaitingStatus("container not yet connectable")
```

Comment:
```suggestion
    def test_given_only_config_storage_container_cant_connect_network_not_available_gocert_not_running_when_collect_unit_status_then_status_is_waiting(
        self, context
    ):
        state = State(
            storage=[Storage(name="config")],
            containers=[Container(name="gocert", can_connect=False)],
            networks={"juju-info": Network([], [], [])},
            leader=True,
        )

        with patch("requests.get", return_value=Mock(side_effect=requests.ConnectionError)):
            out = context.run(Event("collect_unit_status"), state)

        assert out.unit_status == ops.WaitingStatus("container not yet connectable")
```

This comment applies to every collect status related test

Reply:
but these tests are also validating that the configure function does not crash. Should we not check that too?

Reply:
I think we want tests specifically for the collect unit status handler. Now we will also want tests for the configure/config-changed handlers. Those two are independent.

Reply:
How are they independent if the collect unit status handler can't run without a preceding event handler? It doesn't make sense to me to have separate tests to run a function that will never run without another public function that is the actual entry point. It is practically no different than unit testing an internal function which we dont do most of the time.

Regardless, I will touch on this in the spec as it makes sense to talk about it there. I've moved all assert status checks into their own copy of the same test except we only run collect status there.